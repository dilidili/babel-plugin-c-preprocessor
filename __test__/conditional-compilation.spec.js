const loader = require('../src/index')
const { join } = require('path')
const fs = require('fs')

const fixturesDir = join(__dirname, 'fixtures', 'conditional-compilation')

test('single if', () => {
	const actual = loader(fs.readFileSync(join(fixturesDir, 'single-if.js'), 'utf8'))

	expect(actual).toMatchSnapshot()
})

test('if else', () => {
	const actual = loader(fs.readFileSync(join(fixturesDir, 'if-else.js'), 'utf8'))

	expect(actual).toMatchSnapshot()
})
