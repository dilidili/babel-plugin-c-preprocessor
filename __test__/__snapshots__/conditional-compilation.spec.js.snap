// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`elif 1`] = `
"console.log('visible')
"
`;

exports[`if else 1`] = `
"console.log('visible')
"
`;

exports[`single if 1`] = `
"  print(\\"trace message\\");

"
`;
